src/util/std_code.h

```diff
@@ -48,6 +48,26 @@ class codet:public exprt
     set_statement(statement);
   }
 
+  /// \param statement: Specifies the type of the `codet` to be constructed,
+  ///   e.g. `ID_block` for a \ref code_blockt or `ID_assign` for a
+  ///   \ref code_assignt.
+  explicit codet(const irep_idt &statement, operandst &&_op)
+    : exprt(ID_code, empty_typet())
+  {
+    set_statement(statement);
```

Comment:
How about using a delegating constructor rather than duplicating all the work?

Reply:
ok